### PR TITLE
Refactoring create asset and add CI tests to cover

### PR DESCRIPTION
**Pull request by yrong**
Resolves: SNO-817
Requires: https://github.com/Snowfork/polkadot-sdk/pull/99

### CODE REVIEW

File: parachain/primitives/router/src/inbound/mod.rs
Comment (claravanstaden) [inline parachain/primitives/router/src/inbound/mod.rs:303]: I think the pallet index should stay configured in the runtime config. What happens if the ForeignAssets pallet indexes are different per runtime? I think they are the same and that is the convention, but I still think this is runtime related and shouldn't live in the snowbridge crates.

**Review by claravanstaden — COMMENTED** [review]
I had a quick look, I think the polkadot-sdk companion PR should still be opened right? 😄

File: parachain/primitives/router/src/inbound/mod.rs
Comment (yrong) [reply parachain/primitives/router/src/inbound/mod.rs:303]: > What happens if the ForeignAssets pallet indexes are different per runtime? I think they are the same and that is the convention.

Yeah, that's the precondition, checked that all the same in multiple runtimes(e.g. polkadot/kusama). I do not see a reason why they need to break this rule in the future so prefer to make things simple at this moment.

 But I'm open if you all think that's a problem and will make some changes accordingly.

Comment (alistair-singh) [reply parachain/primitives/router/src/inbound/mod.rs:303]: Maybe we can add a test to make sure they are in line?

Comment (alistair-singh) [reply parachain/primitives/router/src/inbound/mod.rs:303]: Ah I see one is added. Ignore this.

**Review by alistair-singh — APPROVED** [review]

File: parachain/primitives/router/src/inbound/mod.rs
Comment (vgeddes) [reply parachain/primitives/router/src/inbound/mod.rs:303]: I agree with Clara here, I am a bit uncomfortable with baking runtime configuration inside our pallets & primitives crate.

Secondly, I am sensitive about changing code unnecessarily at this point. Our next audit is scheduled for Feb 8, and this change will increase the scope of the audit.

I want to reserve any scope increases to accommodate for the second round of code review that the Parity bridges team is going to perform soon.

Comment (yrong) [reply parachain/primitives/router/src/inbound/mod.rs:303]: The initial intention of this PR is for the [comment](https://github.com/paritytech/polkadot-sdk/pull/2522#discussion_r1427682795) to add the call index to some primitives which can be referenced/checked in the AH runtime tests. 

So @vgeddes do we still need this PR?  I think we already have the emulated test register token to cover this use case so maybe the refactoring here and the unit test added in AH may not be necessary at all?

Comment (claravanstaden) [reply parachain/primitives/router/src/inbound/mod.rs:303]: I think Slava wanted us to move those constants here: `polkadot-sdk/bridges/primitives/chain-asset-hub-rococo`. I think it is fine to move the pallet index, but not to our crates. It should still be in the polkadot-sdk, and it should be configured per testnet. I think it can actually go here too: `polkadot-sdk/cumulus/parachains/common/src/rococo.rs`

Comment (yrong) [reply parachain/primitives/router/src/inbound/mod.rs:303]: Thanks Clara and that makes sense!  Then I think it's also better to move the [hard-coded weight](https://github.com/Snowfork/snowbridge/blob/d77201bc370ab2a4da15de701871b4a643bd5b5c/parachain/primitives/router/src/inbound/mod.rs#L191) to some common runtime config.

But if as Vincent mentioned the primary concern is to `reserve any scope increases to accommodate for the second round of code review that the Parity` then perhaps it's not the right time to do the refactoring?

So I'll stop here and let the team decide whether to continue the change or not.

Comment (yrong) [reply parachain/primitives/router/src/inbound/mod.rs]: @claravanstaden Some refactoring in https://github.com/Snowfork/snowbridge/pull/1112/commits/37aa97eedcd0c3a13be72d78603a338ed1f332d0 as you suggested.

File: parachain/primitives/router/src/inbound/mod.rs
Comment (yrong) [inline parachain/primitives/router/src/inbound/mod.rs:97]: So we add a `CallInfo` structure for the create asset and make it runtime config(without any hard coded params anymore), meanwhile we add runtime tests in asset-hub to check the `call_index` and `transact_weight_at_most` all as expected.

**Review by vgeddes — COMMENTED** [review]
Hey Ron this is a very nice improvement. However we have an audit in 2-3 weeks and I want us to have some sort of code freeze for relatively non-important changes.

So lets wait till after the next audit to merge this in.

**Review by alistair-singh — APPROVED** [review]

**Pull request closed without merging**
